remotespark/wrapperkernel/sparkkernelbase.py

```diff
@@ -224,4 +238,4 @@ def _repeat_fatal_error(self):
         error = conf.fatal_error_suggestion().format(self._fatal_error)
         self._logger.error(error)
         self._ipython_display.send_error(error)
-        raise ValueError(self._fatal_error)
+        return self._complete_cell()
```

Comment:
No need to do that. We can just mock what the method that calls IPython returns.

_____________________________
From: msftristew <notifications@github.com<mailto:notifications@github.com>>
Sent: Thursday, January 21, 2016 2:40 PM
Subject: Re: [sparkmagic] Graceful error handling for wrapper kernels (#140)
To: jupyter-incubator/sparkmagic <sparkmagic@noreply.github.com<mailto:sparkmagic@noreply.github.com>>
Cc: Alejandro Guerrero Gonzalez <agg.alex@gmail.com<mailto:agg.alex@gmail.com>>



In remotespark/wrapperkernel/sparkkernelbase.py<https://github.com/jupyter-incubator/sparkmagic/pull/140#discussion_r50476501>:

> @@ -224,4 +238,4 @@ def _repeat_fatal_error(self):>          error = conf.fatal_error_suggestion().format(self._fatal_error)>          self._logger.error(error)>          self._ipython_display.send_error(error)> -        raise ValueError(self._fatal_error)> +        return self._complete_cell()

It's a good idea but I don't personally know what this method returns, since we're offshoring all the work in this method to the IPython kernel. Might require some investigation to see exactly what we should expect in order to add unit tests.

-
Reply to this email directly or view it on GitHub<https://github.com/jupyter-incubator/sparkmagic/pull/140/files#r50476501>.